### ISSUE TITLE
New "ass" subtitle decoding breaks subtitles when seeking

### ISSUE THREAD

**Issue by realnc** [mpv-player/mpv#6970]
Using mpv git master.

Subtitles in MKV files stop showing after seeking forwards (using the up arrow.) Then at some point they start showing again, then they stop, and it's generally erratic behavior.

I bisected the problem to this commit:

1b9370ff92a7fdf347d087db7c29bf682abadb9e

### Log file

https://pastebin.com/raw/W8HzthWk

### Sample files

I'll upload one if the problem cannot be reproduced otherwise.

**Comment by CounterPillow** [comment]
I can't reproduce with [a short sample](https://0x0.st/ztpT.mkv) not a longer sample, can you provide yours? I'm using `mpv 0.29.0-691-gdd547ddcc2` with `--no-config`.

**Comment by bjin** [comment]
I confirm this issue, with srt subtitles in mkv files. Also in my case, in addition to seeking, subtitles may not be displayed from time to time. A workaround is switch to a different subtitle track and switch back.

@CounterPillow could you try a sample with srt subtitles?

**Comment by bjin** [comment]
@CounterPillow I converted the ass subtitle in your sample to [srt format](https://0x0.st/ztpw.mkv), and could confirm that seeking is broken. Reverting the affected commit (with a subsequent commit) fixed the problem for me.

**Comment by jeeb** [comment]
Alright, does this get fixed by using?
```diff
diff --git a/sub/lavc_conv.c b/sub/lavc_conv.c
index 65ebc0d8c7..fd442a11d7 100644
--- a/sub/lavc_conv.c
+++ b/sub/lavc_conv.c
@@ -96,6 +96,8 @@ struct lavc_conv *lavc_conv_create(struct mp_log *log, const char *codec_name,
 #if LIBAVCODEC_VERSION_MICRO >= 100
     avctx->pkt_timebase = avctx->time_base;
     avctx->sub_charenc_mode = FF_SUB_CHARENC_MODE_IGNORE;
+    av_opt_set(lavc_context, "flags2", "+ass_ro_flush_noop",
+               AV_OPT_SEARCH_CHILDREN);
 #endif
     priv->avctx = avctx;
     priv->extradata = talloc_strndup(priv, avctx->subtitle_header,

```

**Comment by jeeb** [comment]
I did test seeking in both MPEG-TS with ARIB captions as well as multiple US CC files, but seems like this is somewhat different :) .

**Comment by jeeb** [comment]
Initially it would look like this is better? The flag basically doesn't reset ReadOrder during flushes.

**Comment by bjin** [comment]
@jeeb seeking seems to be working now. I will test and see if the "missing" subtitles issue is fixed or not.

**Comment by CounterPillow** [comment]
Can reproduce now thanks to bjin and can confirm JEEB's patch works. It's also important to note that playing a file over the network doesn't exhibit the issue presumably due to caching.

**Comment by jeeb** [comment]
Should now be fixed with fc7decd in master. Thank you for everyone for the verification.
